### ISSUE TITLE
Lodash 4.17.0 _.pick breaks when there are dots in keys

### ISSUE THREAD

**Issue by jvah** [lodash/lodash#2809]
With lodash 4.16.6:

```
> _.pick({'foo.txt': 'foo', 'bar.txt': 'bar', 'baz.txt': 'baz'}, ['foo.txt', 'baz.txt'])
{ 'foo.txt': 'foo', 'baz.txt': 'baz' }
```

With lodash 4.17.0:

```
> _.pick({'foo.txt': 'foo', 'bar.txt': 'bar', 'baz.txt': 'baz'}, ['foo.txt', 'baz.txt'])
{ foo: { txt: 'foo' }, baz: { txt: 'baz' } }
```

**Comment by jdalton** [comment]
This is by design. See #723.

**Comment by frakti** [comment]
This issue shouldn't be closed. #723 introduces backward incompatible changes, but only minor version was bumped which is against SemVer.

This will break many production environments when they fetch newest lodash.

**Comment by frakti** [comment]
After the change having issue with `_.omitBy`.

```js
_.omitBy({
  'foo.txt': 'bar', 
  'foo2.txt': 'bar2'
}, (val) => val === 'bar2'
) // { foo: { txt: 'bar' } }
```

**Comment by gkaran** [comment]
I can confirm. This currently breaks our production environment. We cannot build using ember-cli that uses `_.omit`.

**Comment by RuslanZavacky** [comment]
Same thing with ember-cli apps, It seems that now its impossible to publish any ember app to production or start a new one.

@jdalton maybe you can reconsider and revert your change in 4.17 as 4.18 and publish it as 5.0?

**Comment by felixfbecker** [comment]
This should have been introduced as a new function `omitDeep`, like `cloneDeep` etc. It is obvious that this is a breaking change otherwise

**Comment by jdalton** [comment]
@jvah 

I've addressed this issue with #2826 so you won't have to use the an array escape for your use case.
I'll have a v4.17.2 patch release out sometime today.
